whois-nrtm4/src/main/java/net/ripe/db/nrtm4/dao/SnapshotObjectReadOnlyDao.java

```diff
@@ -0,0 +1,39 @@
+package net.ripe.db.nrtm4.dao;
+
+import net.ripe.db.whois.common.dao.jdbc.JdbcStreamingHelper;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.stereotype.Repository;
+
+import javax.sql.DataSource;
+import java.util.function.Consumer;
+
+
+@Repository
+public class SnapshotObjectReadOnlyDao {
+
+    private final JdbcTemplate jdbcTemplate;
+
+    public SnapshotObjectReadOnlyDao(@Qualifier("nrtmSlaveDataSource") final DataSource dataSource) {
+        this.jdbcTemplate = new JdbcTemplate(dataSource);
+    }
+
+    public void snapshotCallbackConsumer(final NrtmSource source, final Consumer<String> fn) {
```

Comment:
rename to something that indicates the functionality, like `readAllObjects` or `getAllObjects` ?

Reply:
Or `consumeAllObjects` ??